src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -199,20 +306,23 @@ uct_rdamcm_cm_ep_set_qp_num(struct rdma_conn_param *conn_param,
     struct ibv_cq *cq;
     ucs_status_t status;
 
-    status = uct_rdmacm_cm_get_cq(uct_rdmacm_cm_ep_get_cm(cep), cep->id->verbs,
-                                  &cq);
-    if (status != UCS_OK) {
-        return status;
-    }
+    status = uct_rdamcm_cm_ep_create_reserved_qpn(cep);
+    if (status == UCS_OK) {
+        conn_param->qp_num = cep->qpn;
+    } else if (status == UCS_ERR_UNSUPPORTED) {
+        status = uct_rdmacm_cm_get_cq(uct_rdmacm_cm_ep_get_cm(cep), cep->id->verbs, &cq);
+        if (status != UCS_OK) {
+            return status;
+        }
 
-    /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
-    status = uct_rdmacm_cm_create_dummy_qp(cep->id, cq, &cep->qp);
-    if (status != UCS_OK) {
-        return status;
+        /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
+        status = uct_rdmacm_cm_create_dummy_qp(cep->id, cq, &cep->qp);
+        if (status == UCS_OK) {
+            conn_param->qp_num = cep->qp->qp_num;
+        }
```

Comment:
let's incapsulate this logic to `uct_rdamcm_cm_ep_create_reserved_qpn`

Reply:
Same as destroy. reserved_qpn is a seperate function than the previous dummy qpn. Do you think we should create another function called uct_rdamcm_cm_ep_create_qpn() and incapsulate above two functions into it?

Reply:
should be aligned with other conversation https://github.com/openucx/ucx/pull/6604#discussion_r613193432

Reply:
@evgeny-leksikov Fixed.